### DISCUSSION TITLE
Not having "Incremental" option for Postgres table has primary key

### DISCUSSION THREAD

**Discussion by hongbo-miao** [airbytehq/airbyte#41966] [Connector Questions] [answered]
We have a table in Postgres: `_airbyte_heartbeat` table

```sql
create table if not exists public._airbyte_heartbeat (
     id serial primary key,
     timestamp timestamptz not null default now()
);
grant insert, update on table public._airbyte_heartbeat to airbyte_user;
```

I expect it has all four options because it has primary key:

- Incremental | Append + Deduped
- Full refresh | Overwrite
- Incremental | Append
- Full refresh | Append

But it only has two options:

![image](https://github.com/user-attachments/assets/1dae6622-2c33-4ebe-ab35-198f0ed91276)

**Comment by johnwasserman** [comment, marked as answer]
@hongbo-miao - for CDC can you confirm you have followed the set up guide correctly:
1. There is an active replication slot
2. The publication configured in Airbyte exists and contains this table
3. The table's `Replica Identity` set to `Default` or `Full` 
4. Run schema discovery in the Airbyte Connection

Full guide is [here](https://docs.airbyte.com/integrations/sources/postgres#setup-using-cdc)

**Reply by hongbo-miao** [reply]
Thanks! @johnwasserman I can confirm:

1. Have replication slot (I am not sure why it is not active, but Airbyte sync well)
![image](https://github.com/user-attachments/assets/20d109fc-e7be-40b7-9a00-1889d83a5654)

2. The publication configured in Airbyte exists and contains `_airbyte_heartbeat` table
![image](https://github.com/user-attachments/assets/e644e3de-45a7-4c20-91a4-e6a49ea95dbf)

3. Replica Identity set to Default.
![image](https://github.com/user-attachments/assets/b1a9669a-2925-49a6-b777-222e58b37142)

I am wondering could it related to underscore `_airbyte_heartbeat`?

**Reply by hongbo-miao** [reply]
Hmm, when I check today, the "Incremental" options of table `_airbyte_heartbeat` show up for all my databases. I am not entirely sure what happened though. Because before I manually synced a few times, it still does not show up. Today when I check it shows up. 🤔

![image](https://github.com/user-attachments/assets/0a8ffb51-12b6-4cab-b143-88bfa1b15b1c)

**Comment by johnwasserman** [comment]
@hongbo-miao - one final thing I forgot to mention initially is that if you had made the changes I suggested, a schema discovery would be required to see the new options appear. It's likely a schema discovery was triggered overnight and now this is visible.

**Reply by hongbo-miao** [reply]
Could be, thanks @johnwasserman !
